Azure gateway spin-ups

`test_replicator_client.py` has now been modified to read azure subscription from OS variable similar to the `cli`. It has an option to override the OS if a subscription has been input from the command line --azure-subscription when invoking test_replicator. So, to default back and read from OS variable, we need to not input the subscription. args.azure-subscription then defaults to None. However, this raises the following issue. 


`2022-01-28 02:54:38.316 | INFO     | __main__:main:178 - Provisioning gateway instances
Traceback (most recent call last):
  File "skylark/test/test_replicator_client.py", line 206, in <module>
    main(parse_args())
  File "skylark/test/test_replicator_client.py", line 179, in main
    rc.provision_gateways(
  File "/home/ubuntu/skylark/skylark/replicate/replicator_client.py", line 76, in provision_gateways
    assert len(azure_regions_to_provision) ==0, "Azure not enabled"
AssertionError: Azure not enabled`

Command to replicate:
`python skylark/test/test_replicator_client.py --gateway-docker-image $SKYLARK_DOCKER_IMAGE --n-chunks 100 --chunk-size-mb 4 --num-gateways 1 --src-region azure:eastus --dest-region azure:westus  --gcp-project skylark-shishir --bucket-prefix t11`

Commit ID: f32b4db3e2600a1f5581336f0bca43f8af5a8dc9 of #100 

Workaround for now: just include `--azure-subscription`

--- a/skylark/test/test_replicator_client.py
+++ b/skylark/test/test_replicator_client.py
@@ -65,6 +65,11 @@ def parse_args():
 
 
 def main(args):
+    config = load_config()
+    gcp_project = args.gcp_project or config.get("gcp_project_id")
+    azure_subscription = args.azure_subscription or config.get("azure_subscription_id")
+    logger.debug(f"Loaded gcp_project: {gcp_project}, azure_subscription: {azure_subscription}")
+
     src_bucket = f"{args.bucket_prefix}-skylark-{args.src_region.split(':')[1]}"
     dst_bucket = f"{args.bucket_prefix}-skylark-{args.dest_region.split(':')[1]}"
     obj_store_interface_src = ObjectStoreInterface.create(args.src_region, src_bucket)
@@ -128,14 +133,10 @@ def main(args):
     logger.info("Creating replication client")
 
     # Getting configs
-    config = load_config()
-    gcp_project = args.gcp_project or config.get("gcp_project_id")
-    azure_subscription = args.azure_subscription or config.get("azure_subscription_id")
-    logger.debug(f"Loaded gcp_project: {gcp_project}, azure_subscription: {azure_subscription}")
     rc = ReplicatorClient(
         topo,
-        gcp_project=args.gcp_project,
-        azure_subscription=args.azure_subscription,
+        gcp_project=gcp_project,
+        azure_subscription=azure_subscription,
         gateway_docker_image=args.gateway_docker_image,
         aws_instance_class=args.aws_instance_class,
         azure_instance_class=args.azure_instance_class,

